[Infra] Cloudflare Routes API Error 10013 Causes False Deployment Failures

## Priority
P2 - Low (deployments work despite this error)

## Description

The GitHub Actions `Deploy to Production` workflow reports failure even though deployments succeed. This is due to a Cloudflare API error 10013 when configuring routes.

## Logs

```
✨ Success! Uploaded 46 files (4.56 sec)
Uploaded streaming-patterns-staging (11.86 sec)

✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/streaming-patterns-staging/routes) failed.

  An unknown error has occurred. [code: 10013]
```

## Impact

- Workflow shows red ❌ even though deployment succeeds
- Confusing for developers
- May block merges if branch protection requires passing checks

## Root Cause

The routes API call fails when:
1. Routes are already configured for the worker
2. The worker is deployed to a zone with existing route bindings
3. Possible rate limiting or API inconsistency from Cloudflare

## Proposed Fix Options

1. **Add continue-on-error for routes** - Let the workflow succeed if upload works
2. **Skip routes call for workers_dev** - Staging uses workers.dev, no custom routes needed
3. **Check existing routes first** - Only call routes API if configuration changed
4. **Upgrade wrangler** - v4.51.0 may have fixes (currently using v3.90.0)

## References

- https://github.com/cloudflare/workers-sdk/issues (search for error 10013)
- Related to issue #183

## Files to Modify

- `.github/workflows/deploy-production.yml`
- `wrangler.toml` (possibly remove routes from staging env)

--- a/wrangler.toml
+++ b/wrangler.toml
@@ -32,5 +32,8 @@ ENVIRONMENT = "production"
 [env.staging]
 name = "streaming-patterns-staging"
 workers_dev = true
+# Explicitly disable routes for staging (uses workers.dev subdomain)
+# This prevents Cloudflare API error 10013 when routes are inherited from parent config
+routes = []
 vars = { ENVIRONMENT = "staging" }
 

